### ISSUE TITLE
Signup alerts

### ISSUE THREAD

**Issue by HipyCas** [HipyCas/WikiGo-Bootstrap#11]
When registering, after clicking the submit button, even if the registration is successful, the password validation method raises errors/alerts, that stack on top of the successful register alert. The code snippets are shown below.
*This issue was introduced to the master branch with pull request #10 / commit 0af3def21836b0d52f39ea102950fad092c9041f*

#### Register handle
```go
func registerHandle(w http.ResponseWriter, r *http.Request) {
	if r.Method == "GET" {
		renderTemplate(w, "register")
	} else {
		r.ParseForm()
		if r.Form.Get("password") != r.Form.Get("passwordRepeat") {
			log.Printf("The passwords %s and %s do not match", r.Form.Get("password"), r.Form.Get("repeatPassword"))
			addAlertCreate(5, "The passwords do match")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusFound)
		}
		valid, message := isValidPassword(r.Form.Get("password"))
		if !valid {
			log.Printf("The password %s is not valid: %s", r.Form["password"][0], message)
			addAlertCreate(5, message)
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusFound)
			return
		}
		_, err := os.Open("user/" + r.Form.Get("username") + ".xml")
		if err == nil {
			log.Printf("Username %s already exists: %v", r.Form.Get("username"), err)
			addAlertCreate(5, "Username not available")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusFound)
			return
		}
		user := User{Username: r.Form.Get("username") /*FirstName: r.Form.Get("firstName"), LastName: r.Form.Get("lastName"),*/, Email: r.Form.Get("email") /*, PhoneNumber: r.Form.Get("phoneNumber"), Country: r.Form.Get("country")*/}
		passwordHash, err := bcrypt.GenerateFromPassword([]byte(r.Form.Get("password")), bcrypt.DefaultCost)
		if err != nil {
			log.Printf("Error when hasing password %s: %v", r.Form.Get("password"), err)
			log.Fatal(err)
			addAlertCreate(5, "Internal server error while registering")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusFound)
			return
		}
		user.Password = passwordHash
		out, err := xml.MarshalIndent(user, " ", "\t")
		if err != nil {
			log.Printf("Error when generating XML for user %v: %v", user, err)
			addAlertCreate(5, "Internal server error while registering")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusInternalServerError)
			return
		}
		_, err = os.Stdout.Write([]byte(xml.Header))
		if err != nil {
			log.Printf("Error when writing XML header to console for user %v: %v", user, err)
			addAlertCreate(5, "Internal server error while registering")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusInternalServerError)
			return
		}
		_, err = os.Stdout.Write(out)
		if err != nil {
			log.Printf("Error when writing XML to console for user %v: %v", user, err)
			addAlertCreate(5, "Internal server error while registering")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusInternalServerError)
			return
		}
		err = ioutil.WriteFile("user/"+user.Username+".xml", out, os.ModePerm)
		if err != nil {
			log.Printf("Error when wsaving XML to file user/%s.xml for user %v: %v", user.Username, user, err)
			addAlertCreate(5, "Internal server error while registering")
			r.Header.Set("Method", "GET")
			http.Redirect(w, r, "/register/", http.StatusInternalServerError)
			return
		}
		addAlertCreate(3, "Succesfully created account!")
		addAlertCreate(2, "Login now to access it")
		http.Redirect(w, r, "/login/", http.StatusFound)
	}
}
```
### Password validation
```go
func isValidPassword(password string) (ok bool, msg string) {
	var (
		upp, low, num bool
		tot           int
	)
	for _, char := range password {
		switch {
		case unicode.IsUpper(char):
			upp = true
			tot++
		case unicode.IsLower(char):
			low = true
			tot++
		case unicode.IsDigit(char):
			num = true
			tot++
			/*
				case unicode.IsSymbol(char):
					sym = true
					tot++
				default:
					return false, "Invalid character on password (only unicode characaters accepted)"*/
		}
	}

	if !upp {
		msg = "Password must include at least one uppercase letter"
		ok = false
	} else if !low {
		msg = "Password must include at least one lowercase letter"
		ok = false
	} else if !num {
		msg = "Password must include at least one digit/number"
		ok = false
		/*} else if !sym {
		msg = "Password must include at least one symbol"
		ok = false*/
	} else if tot < 8 {
		msg = "Password must be longer than 8"
		ok = false
	} else {
		msg = ""
		ok = true
	}
	return
}
```
#### Alert struct & methods
```go
type Alert struct {
	Level     int
	Primary   bool
	Secondary bool
	Info      bool
	Success   bool
	Warning   bool
	Danger    bool
	Light     bool
	Dark      bool
	Msg       string
}

var alerts []Alert

func addAlert(alert Alert) error {
	if alert.Level < 0 || alert.Level > 7 {
		return &InvalidAlertLevel{Lvl: alert.Level}
	}
	alerts = append(alerts, alert)
	return nil
}

func addAlertCreate(lvl int, msg string) (Alert, error) {
	a := Alert{Level: lvl, Msg: msg}
	switch lvl {
	case 0:
		a.Primary = true
	case 1:
		a.Secondary = true
	case 2:
		a.Info = true
	case 3:
		a.Success = true
	case 4:
		a.Warning = true
	case 5:
		a.Danger = true
	case 6:
		a.Light = true
	case 7:
		a.Dark = true
	default:
		return Alert{}, &InvalidAlertLevel{Lvl: lvl}
	}
	alerts = append(alerts, a)
	return a, nil
}

func getAlerts() []Alert {
	toReturn := []Alert{}
	for i := 0; i < len(alerts); i++ {
		log.Printf("Got alert of msg: %s and level %d", alerts[i].Msg, alerts[i].Level)
		toReturn = append(toReturn, alerts[i])
	}
	for j := 0; j < len(alerts); j++ {
		removeAlert(j)
	}
	return toReturn
}

func removeAlert(index int) {
	alerts[index] = alerts[len(alerts)-1]
	alerts = alerts[:len(alerts)-1]
}
```

https://app.leanboard.io/board/2ae0221b-43be-4c16-95b0-be8c867c0fee

**Comment by HipyCas** [comment]
Issue fixed on commit 7e7173909dcdae4ca5a215c8c522a5c47fc50133 from pull request #13

**Comment by HipyCas** [comment]
Issue fixed on commit 7e7173909dcdae4ca5a215c8c522a5c47fc50133 from pull request #13
